netmap/netmap_contract.go

```diff
@@ -13,25 +13,36 @@ import (
 	"github.com/nspcc-dev/neofs-contract/common"
 )
 
-type (
-	storageNode struct {
-		info []byte
-	}
+// NodeState is an enumeration for node states.
+type NodeState int
 
-	netmapNode struct {
-		node  storageNode
-		state NodeState
-	}
+// Various Node states
+const (
+	_ NodeState = iota
 
-	// NodeState is an enumeration for node states.
-	NodeState int
+	// NodeStateOnline stands for nodes that are in full network and
+	// operational availability.
+	NodeStateOnline
 
-	record struct {
-		key []byte
-		val []byte
-	}
+	// NodeStateOffline stands for nodes that are in network unavailability.
+	NodeStateOffline
+
+	// NodeStateMaintenance stands for nodes under maintenance with partial
+	// network availability.
+	NodeStateMaintenance
 )
 
+// Node groups data related to NeoFS storage nodes registered in the NeoFS
+// network. The information is stored in the current contract.
+type Node struct {
```

Comment:
If you're planning to update contract in the existing chain, then you need to write a migration routine. It should migrate the old netmap contract storage scheme to the new one (storageNode -> Node transformation), otherwise contract won't be able to work with the previously saved data.

Reply:
Yes, migration job should be performed, thanks.

I've started from documenting (for developers)  the storage structure #276.

Reply:
We've decided to postpone the migration #277.